File: tests/src/testTransport/PassiveLink.cpp
#include "PassiveLink.h"

#include "Transport.h"

namespace testTransport {

PassiveLink::PassiveLink( Transport* creator, const std::string& address ) : _address( address ), 
    _creator( creator )
{
}

PassiveLink::PassiveLink()
{
}
    
PassiveLink::~PassiveLink()
{
    _creator->onLinkDestructor( _address );
}

// ------ reef.IPassiveLink Methods ------ //

bool PassiveLink::receive( std::string& msg )
{
    return _creator->check( _address, msg );
}

void PassiveLink::sendReply( const std::string& msg )
{
    _creator->postReply( _address, msg );
}

CORAL_EXPORT_COMPONENT( PassiveLink, PassiveLink );

} // namespace reef

File: tests/modules/testTransport/PassiveLink.lua
local PassiveLink = co.Component( { name = "testTransport.PassiveLink", 
								provides = { passive = "reef.IPassiveLink" } } )

function PassiveLink:sendReply( msg )
	self.net:sendReply( self.address, msg )
end

function PassiveLink:receive()
	local msg = self.net:check( self.address )
	--print( "PassiveLink of address ".. self.address .. " received: " .. tostring( msg ) )
	
	if msg then
		return true, msg
	end
	
	return false, ""
end

function PassiveLink:getAddress()
	return self.address
end

return PassiveLink